File: mobile_app/app/src/main/java/com/smartambulance/driver/mqtt/MqttTopics.kt
package com.smartambulance.driver.mqtt

object MqttTopics {
    fun ambulanceLoRaGps(ambulanceId: String) = "smart-ambulance/ambulances/$ambulanceId/lora-gps"
    fun junctionApproach(junctionId: String) = "smart-ambulance/junctions/$junctionId/approach"
    fun ambulanceStatus(ambulanceId: String) = "smart-ambulance/ambulances/$ambulanceId/status"
    fun tripEvents(tripId: String) = "smart-ambulance/trips/$tripId/events"
    fun junctionEvents(junctionId: String) = "smart-ambulance/junctions/$junctionId/events"
    fun junctionSignal(junctionId: String) = "smart-ambulance/junctions/$junctionId/signal"
}

File: dashboard/src/integrations/mqttTopics.js
export const mqttTopics = {
  ambulanceLoRaGps: (ambulanceId) => `smart-ambulance/ambulances/${ambulanceId}/lora-gps`,
  junctionApproach: (junctionId) => `smart-ambulance/junctions/${junctionId}/approach`,
  junctionEvents: (junctionId) => `smart-ambulance/junctions/${junctionId}/events`,
  junctionSignal: (junctionId) => `smart-ambulance/junctions/${junctionId}/signal`,
  ambulanceStatus: (ambulanceId) => `smart-ambulance/ambulances/${ambulanceId}/status`,
  tripEvents: (tripId) => `smart-ambulance/trips/${tripId}/events`,
};

export const mqttBrokerDefaults = {
  localWebSocketUrl: "ws://localhost:9001",
  publicTestWebSocketUrl: "wss://broker.hivemq.com:8884/mqtt",
};
